File: demo/src/main/java/com/training/demo/dto/response/Payment/PaymentResponse.java
package com.training.demo.dto.response.Payment;

import com.training.demo.utils.enums.PaymentMethod;
import com.training.demo.utils.enums.PaymentStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PaymentResponse {
    private Long id;
    private Long orderId;
    private String orderNumber;
    private PaymentMethod paymentMethod;
    private BigDecimal amount;
    private String transactionId;
    private PaymentStatus status;
    private LocalDateTime paymentDate;
    private String paymentInfo;
    private String errorMessage;
    private LocalDateTime createdAt;
}

File: src/app/core/models/response/Payment/PaymentResponse.ts
import { PaymentMethod } from "src/app/utils/PaymentMethod";
import { PaymentStatus } from "src/app/utils/PaymentStatus";

export interface PaymentResponse {
  id: number;
  orderId: number;
  orderNumber: string;
  paymentMethod: PaymentMethod;
  amount: number;
  transactionId?: string;
  status: PaymentStatus;
  paymentDate?: string;
  paymentInfo?: string;
  errorMessage?: string;
  createdAt: string;
}
